product_name_generator/models/product_template.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    openai_generated_name = fields.Char()
+
+    def generate_name(self, session, is_published):
+        base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
+        input_image = f"{base_url}/web/image/{self._name}/{self.id}/image_1920"
+        input_datas = json.dumps(
+            [
+                {
+                    "role": "user",
+                    "content": [{"type": "input_image", "image_url": input_image}],
+                }
+            ]
+        )
+        response_json_str = session.call_openAI(input_datas)
+        response_json = json.loads(response_json_str)
+        product_name = response_json.get("product_name")
+        self.openai_generated_name = product_name
+        self.is_published = is_published
+
+    def action_generate_product_name(self):
+        session = self.env.ref(
+            "product_name_generator.openai_vision_session_product_name_generator"
+        )
+        for rec in self:
+            if not rec.image_1920:
+                raise UserError(_("Please upload the image first."))
+            is_published = rec.is_published
+            if not is_published:
+                # Need to be published to be accessible to the public.
+                rec.is_published = True
```

Comment:
If is_published was set to False, it is temporarily set to True. But shouldn't it be reverted back to False ?

Reply:
Yes. I was done in `def generate_name()`.